plugins/utils/unified-toc.js

```
@@ -0,0 +1,27 @@
+const { load } = require('js-toml');
+const fs = require('fs/promises');
+
+const createTocNodes = async ({ createNode, createNodeId, createContentDigest }) => {
+  // Get all MongoDB products for the sidenav
+
+  try {
+    const tomlContents = (await fs.readFile(`${process.cwd()}/toc_data/toc.toml`)).toString();
+    const toc = load(tomlContents);
+
+    createNode({
+      tocTree: toc,
+      id: createNodeId('toc'),
+      internal: {
+        contentDigest: createContentDigest(toc),
+        type: 'TOC',
+      },
+      parent: null,
+    });
+  } catch (e) {
+    console.error('error occurred when reading the toc.toml', e);
```

minor,  do we want to throw the same error and stop build here? would imagine this (toc node creation) is essential for site to build

---

Thanks seung, made that change. We wont be merging this into main and just leaving it on this branch. Will be making  a ticket for the future on refining the toc.toml file based on writer feedback!